presto-main/src/main/java/com/facebook/presto/sql/planner/PartitioningScheme.java

```diff
@@ -22,17 +22,54 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.function.IntPredicate;
 
+import static com.facebook.presto.sql.planner.PartitioningScheme.Replication.REPLICATE_NOTHING;
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class PartitioningScheme
 {
+    public enum Replication
+    {
+        REPLICATE_NOTHING(
```

Comment:
Also, how does this relate to https://github.com/prestodb/presto/pull/5323? (Where we try to capture the notion of a value that is not equal to itself for the purpose of deciding whether to replicate it or not)